src/tests/integration/mapping.test.ts

```diff
@@ -67,6 +67,45 @@ describe('/api', () => {
     const mappingData = await getMapping.json()
     expect(Object.keys(mappingData).length).toEqual(0)
   })
+  
+  it('checks if changes to the resource has been saved', async () => {
+    const subDomain = `testing${uuidv4()}`
+    const domain = 'integration'
+    const port = '3457'
+    const createMapping = await mappingAdapter('/', 'POST', {
+      domain, 
+      subDomain,
+      port
+    })
+    expect(createMapping.status).toEqual(200)
+    const mapping = await createMapping.json()
+    const patchMapping = await mappingAdapter(`/edit/${mapping.id}`, 'PATCH', {
+      subDomain: 'testingPatch',
```

Comment:
this shhould actually thhrow an error instead of succeeding. you cannot crreate 2 of the same domains. For example:

```
kalra.diva.club exists
// Scenaio 1: A crreate mapping request to kalra.diva.club shhould fail.
// Scenario 2: An edit of existing mapping rrequest to kalra.diva.club shhould also fail. There shhould never exist 2 domains taht arre the same
```

sorry for the typos, keyboards on thhe macbook is so messed up. Double chharracters, keys sometimes not registering, and a broken command key. Do not buy a macbook.... lol.